src/generate-types.ts

```diff
@@ -5,230 +5,52 @@ import * as _ from 'lodash';
 import * as assert from 'assert';
 import * as oautil from './util';
 import { UnsupportedFeatureBehaviour } from './util';
```

Comment:
I wrapped the whole function to a closure to get some asynchronicity safe state going on in `state` and `options` hence the big diff. The actual changes are in the top part of `run`